src/hooks/useOrder.tsx

```diff
@@ -0,0 +1,52 @@
+import { useState } from "react"
+import axios from "axios"
+
+export interface OrderRequest {
+  productId: number
+  message: string
+  messageCardId: string
+  ordererName: string
+  receivers: {
+    name: string
+    phoneNumber: string
+    quantity: number
+  }[]
+}
+
+export function useOrder() {
+  const [orderLoading, setLoading] = useState(false)
+  const [orderError, setError] = useState<null | string>(null)
+  const [success, setSuccess] = useState(false)
+  const baseUrl = import.meta.env.VITE_BASE_URL
+  const orderUrl = new URL(`/api/order`, baseUrl).toString()
+
+  const createOrder = async (order: OrderRequest, token: string) => {
+    setLoading(true)
+    setError(null)
+    setSuccess(false)
+    try {
+      const res = await axios.post(orderUrl, order, {
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: token,
+        },
+      })
+      setSuccess(true)
+      return res.data
+    } catch (err: any) {
+      const errorMessage = err?.response?.data?.message || "주문 실패"
+      setError(errorMessage)
+      
+      if (err?.response?.status === 401) {
+        const error401 = new Error(errorMessage)
+        error401.name = "401"
+        throw error401
+      }
+      throw err
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  return { createOrder, orderLoading, orderError, success }
+}
```

Comment:
[제안] useFetch와 별개로 각 post, put 등의 mutation 요청의 상태를 자동으로 리액트와 동기화해주는 useMutation 훅으로 mutation 요청의 핵심 로직만 분리하면, 주문 처리 훅, 로그인 훅 등에서 재사용하기 좋을 것 같네요!
```suggestion
export function useOrder() {
  const { mutate, isMutating, error, isSuccess} = useMutation(orderUrl);
  const baseUrl = import.meta.env.VITE_BASE_URL
  const orderUrl = new URL(`/api/order`, baseUrl).toString()

  const createOrder = async (order: OrderRequest, token: string) => {
    const res = await mutate(order, {
      headers: {
        "Content-Type": "application/json",
        Authorization: token,
      },
    })

    return res.data;
  }

  return { createOrder, orderLoading: isMutating, orderError: error, success: isSuccess }
}
```

Reply:
이 부분은 3단계를 할때쯤에 서둘러 구현 후 리팩토링해보겠습니다! 피드백 감사합니다!